src/autoscaler/sync/dblock.go

```diff
@@ -0,0 +1,79 @@
+package sync
+
+import (
+	"autoscaler/db"
+	"autoscaler/models"
+	"os"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+
+	"github.com/tedsuo/ifrit"
+)
+
+type DatabaseLock struct {
+	logger lager.Logger
+}
+
+func NewDatabaseLock(logger lager.Logger) *DatabaseLock {
+	return &DatabaseLock{
+		logger: logger,
+	}
+}
+
+func (dblock *DatabaseLock) InitDBLockRunner(ttl time.Duration, owner string, lockDB db.LockDB) ifrit.Runner {
+	dbLockMaintainer := ifrit.RunFunc(func(signals <-chan os.Signal, ready chan<- struct{}) error {
+		lockTicker := time.NewTicker(ttl)
+		readyFlag := true
```

Comment:
a little bit misleading, perhaps initialize with false, when it is ready then assign true value

Reply:
the readyFlag is being used to indicate whether it's ready to get the lock, initially set to true to indicate that the instance is ready to get the lock, as soon as it got the lock we made it to false.

Reply:
it is confusing since there is also ready channel while it has different meaning... If so can we change the variable name?